File: 백준/Gold/9527. 1의 개수 세기/1의 개수 세기.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {

    private final BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
    private final BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
    private final StringBuilder result = new StringBuilder();

    public void solve() throws IOException {
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        long from = Long.parseLong(tokenizer.nextToken());
        long to = Long.parseLong(tokenizer.nextToken());
        result.append(count(to) - count(from - 1));
    }

    private long count(long num) {
        long count = 0;
        int len = Long.toString(num, 2).length();
        for (long i = 1; i <= len; i++) {
            long ca = ((num + 1) / (long) Math.pow(2, i)) * (long) Math.pow(2, i - 1);
            long cb = ((num + 1) % (long) Math.pow(2, i)) - (long) Math.pow(2, i - 1);
            count += ca + Math.max(cb, 0);
        }
        return count;
    }

    private void finish() throws IOException {
        writer.write(result.toString());
        writer.flush();
        writer.close();
    }

    public static void main(String[] args) throws IOException {
        Main main = new Main();
        main.solve();
        main.finish();
    }
}

File: 백준/Gold/9527. 1의 개수 세기/1의 개수 세기.kt
import kotlin.math.max
import kotlin.math.pow

class Main(private val result: StringBuilder = StringBuilder()) {

    fun solve() {
        val (from, to) = readln().split(" ").map { it.toLong() }
        result.append(count(to) - count(from - 1))
    }

    private fun count(num: Long): Long {
        var count: Long = 0
        val len = num.toString(2).length
        for (i in 1..len) {
            val ca = (num + 1) / 2.0.pow(i).toLong() * 2.0.pow(i - 1).toLong()
            val cb = (num + 1) % 2.0.pow(i).toLong() - 2.0.pow(i - 1).toLong()
            count += ca + max(cb, 0)
        }
        return count
    }

    fun finish(): Unit = with(System.out.bufferedWriter()) {
        write(result.toString())
        flush()
        close()
    }
}

fun main() {
    val main = Main()
    main.solve()
    main.finish()
}